pkg/controller/controller.go

```diff
@@ -352,6 +363,20 @@ shutdown:
 	close(c.terminated)
 }
 
+func (c *controller) needsDelay(minInterval time.Duration) (bool, time.Duration) {
+	if minInterval == time.Duration(0) {
+		return false, 0
+	}
```

Comment:
nit: `time.Sleep(time.Duration(0))` returns immediately, so we can simplify by just skipping the bool.

```suggestion
func (c *controller) needsDelay(minInterval time.Duration) (time.Duration) {
	if minInterval == time.Duration(0) {
		return 0
	}
```